src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -146,28 +112,65 @@ object KustoDataSourceUtils{
       keyVaultUri != ""}){
       // KeyVault Authentication
       var keyVaultAppId: String = null
+      var kustoAuthentication: KustoAuthentication = authentication
 
       if({keyVaultAppId = parameters.getOrElse(KustoOptions.KEY_VAULT_APP_ID, "")
```

Comment:
IMO much more readable if keyVaultAppId = parameters.getOrElse(KustoOptions.KEY_VAULT_APP_ID, "" is outside the if()